contracts/service-registry/src/state.rs

```diff
@@ -173,7 +211,17 @@ pub fn deregister_chains_support(
     verifier: VerifierAddress,
 ) -> Result<(), ContractError> {
     for chain in chains {
-        VERIFIERS_PER_CHAIN.remove(storage, (&service_name, &chain, &verifier));
+        VERIFIERS_PER_CHAIN_INDEXED_MAP
+            .remove(
+                storage,
+                (service_name.clone(), chain.clone(), verifier.clone()),
+            )
+            .unwrap_or_else(|_| {
+                panic!(
+                    "Failed to deregister support for chain {:?} from verifier {:?}",
+                    chain, verifier
+                )
+            });
```

Comment:
I wouldn't panic here, just return the error. Use change_context if need be

Reply:
I used `map_err`.